config/jobs/kubernetes/test-infra/test-infra-presubmits.yaml

```diff
@@ -129,6 +129,47 @@ presubmits:
           limits:
             cpu: "1"
             memory: "2Gi"
+  - name: pull-test-infra-prow-checkconfig-deprecated-job-configs
+    cluster: eks-prow-build-cluster
+    optional: true
+    decorate: true
+    run_if_changed: '^(config/prow/(config|plugins).yaml$|config/jobs/.*.yaml$)'
+    spec:
+      containers:
+      - image: gcr.io/k8s-prow/checkconfig:v20230919-a4926a4d02
+        command:
+        - sh
+        args:
+        - -c
+        - |
+          set -o errexit
+          set -o nounset
+          set -o pipefail
+          set -o xtrace
+
+          HEAD_CHECK=$(checkconfig \
+            --config-path=config/prow/config.yaml \
+            --job-config-path=config/jobs \
+            --plugin-config=config/prow/plugins.yaml \
+            --warnings=non-decorated-jobs
+
+          git checkout master
+          MASTER_CHECK=$(checkconfig \
+            --config-path=config/prow/config.yaml \
+            --job-config-path=config/jobs \
+            --plugin-config=config/prow/plugins.yaml \
+            --warnings=non-decorated-jobs
+           
+          echo $MASTER_CHECK > /tmp/master
+          echo $HEAD_CHECK > /tmp/head
+          diff /tmp/master /tmp/head
```

Comment:
This will fail also when the number of jobs without decoration *decreases*. Having a job that throws an error for PRs which do the right thing is going to be confusing.

Reply:
you mean in a PR that removes non-decorated jobs?
the job is optional and the goal is to not have any and become this strict, I don't know how far we are though